docs/src/literate/quasi_incompressible_hyperelasticity.jl

```diff
@@ -0,0 +1,362 @@
+function solve(interpolation_u, interpolation_p)
+    reset_timer!()
+
+    ## import the mesh
+    grid = importTestGrid()
+
+    ## Material parameters
+    μ = 1.
+    λ = 1.E4 * μ
+    mp = NeoHooke(μ, λ)
+
+    ## Create the DofHandler and CellValues
+    dh = create_dofhandler(grid, interpolation_u, interpolation_p)
+    cellvalues_u, cellvalues_p, facevalues_u = create_values(interpolation_u, interpolation_p)
+
+    ## Create the DirichletBCs
+    dbc = create_bc(dh)
+
+    ## Pre-allocation of vectors for the solution and Newton increments
+    _ndofs = ndofs(dh)
+    w  = zeros(_ndofs)
+    ΔΔw = zeros(_ndofs)
+    apply!(w, dbc)
+
+    ## Create the sparse matrix and residual vector
+    K = create_sparsity_pattern(dh)
+    f = zeros(_ndofs)
+
+    ## We run the simulation parameterized by a time like parameter. `Tf` denotes the final value
+    ## of this parameter, and Δt denotes its increment in each step
+    Tf = 2.0;
+    Δt = 0.1;
+    NEWTON_TOL = 1e-8
+
+    pvd = paraview_collection("hyperelasticity_incomp_mixed.pvd");
+    for t ∈ 0.0:Δt:Tf
+        ## Perform Newton iterations
+        print("Time is $t")
+        Ferrite.update!(dbc, t)
+        apply!(w, dbc)
+        newton_itr = -1
+        prog = ProgressMeter.ProgressThresh(NEWTON_TOL, "Solving:")
+        fill!(ΔΔw, 0.0);
+        while true; newton_itr += 1
+            assemble_global!(K, f, cellvalues_u, cellvalues_p, dh, mp, w)
+            norm_res = norm(f[Ferrite.free_dofs(dbc)])
+            apply_zero!(K, f, dbc)
+            ProgressMeter.update!(prog, norm_res; showvalues = [(:iter, newton_itr)])
+
+            if norm_res < NEWTON_TOL
+                break
+            elseif newton_itr > 30
+                error("Reached maximum Newton iterations, aborting")
+            end
+            ## Compute the incremental `dof`-vector (both displacement and pressure)
+            ΔΔw .= K\f;
+
+            apply_zero!(ΔΔw, dbc)
+            w .-= ΔΔw
+        end;
+
+        ## Save the solution fields
+        @timeit "export" begin
+            vtk_grid("hyperelasticity_incomp_mixed_$t.vtu", dh) do vtkfile
+                vtk_point_data(vtkfile, dh, w)
+                vtk_save(vtkfile)
+                pvd[t] = vtkfile
+            end
+        end
+
+        print_timer(title = "Analysis with $(getncells(grid)) elements", linechars = :ascii)
+    end;
+    vtk_save(pvd);
+    vol_def = calculate_volume_deformed_mesh(w, dh, cellvalues_u);
+    print("Deformed volume is $vol_def")
+    return vol_def;
+end;
+
+# We can now test the solution using the Taylor-Hood approximation
+quadratic = Lagrange{3, RefTetrahedron, 2}()
+linear = Lagrange{3, RefTetrahedron, 1}()
+vol_def = solve(quadratic, linear);
+
+# We can also check that the deformed volume is indeed close to 1 (as should be for a nearly incompressible material)
+using Test                #src     
+@test isapprox(vol_def, 1.0, atol=1E-3) # src
```

Comment:
Ninja space :) `# src` does not work, it must be `#src` as in the line before.

Reply:
Fixed